src/neptune/internal/operation_processors/async_operation_processor.py

```diff
@@ -149,23 +165,11 @@ def _check_no_progress(self):
                 self._async_no_progress_callback()
                 self._should_call_no_progress_callback = False
 
-    def flush(self):
-        self._queue.flush()
-
-    def start(self):
-        self._consumer.start()
-
-    def pause(self):
-        self._consumer.pause()
-        self._queue.flush()
-
-    def resume(self):
-        self._consumer.resume()
+    def _wait_for_queue_empty(self, initial_queue_size: int, seconds: Optional[float]) -> None:
+        waiting_start: float = monotonic()
+        time_elapsed: float = 0.0
+        max_reconnect_wait_time: float = self.STOP_QUEUE_MAX_TIME_NO_CONNECTION_SECONDS if seconds is None else seconds
```

Comment:
Is `seconds == 0.` a legal setting?

If yes, then I have no comments.

If no, it can be simplified to
```
max_reconnect_wait_time: float = seconds or self.STOP_QUEUE_MAX_TIME_NO_CONNECTION_SECONDS
```

Reply:
Probably it's allowed. We would like to apply `self.STOP_QUEUE_MAX_TIME_NO_CONNECTION_SECONDS` on whenever `seconds is None`.

Reply:
Probably it is allowed. Something like immediate stop/wait.

Reply:
So we would like to apply `self.STOP_QUEUE_MAX_TIME_NO_CONNECTION_SECONDS` only to overcome the `Optional` specification :wink:.